rbx_xml/src/types/font.rs

```diff
@@ -0,0 +1,199 @@
+    convert::TryInto,
+    io::{Read, Write},
+};
+
+use rbx_dom_weak::types::{Font, FontStyle, FontWeight};
+
+use crate::{
+    core::XmlType,
+    deserializer_core::{XmlEventReader, XmlReadEvent},
+    error::{DecodeError, DecodeErrorKind, EncodeError},
+    serializer_core::{XmlEventWriter, XmlWriteEvent},
+};
+
+fn write_content(
+    writer: &mut XmlEventWriter<impl Write>,
+    content: &str,
+    tag: &str,
+) -> Result<(), EncodeError> {
+    writer.write(XmlWriteEvent::start_element(tag))?;
+    if content.is_empty() {
+        // This doesn't feel like a great XML idiom
+        writer.write(XmlWriteEvent::start_element("null"))?;
+    } else {
+        writer.write(XmlWriteEvent::start_element("url"))?;
+        writer.write_string(content)?;
+    }
+    writer.write(XmlWriteEvent::end_element())?;
+    writer.write(XmlWriteEvent::end_element())?;
+    Ok(())
+}
+
+fn read_content_inner(reader: &mut XmlEventReader<impl Read>) -> Result<String, DecodeError> {
+    match reader.expect_next()? {
+        XmlReadEvent::StartElement {
+            name,
+            attributes,
+            namespace,
+        } => match name.local_name.as_str() {
+            "null" => {
+                reader.expect_end_with_name("null")?;
+                Ok(String::new())
+            }
+            "url" => {
+                let value = reader.read_characters()?;
+                reader.expect_end_with_name("url")?;
+                Ok(value)
+            }
+            _ => {
+                let event = XmlReadEvent::StartElement {
+                    name,
+                    attributes,
+                    namespace,
+                };
+                Err(reader.error(DecodeErrorKind::UnexpectedXmlEvent(event)))
+            }
+        },
+        event => Err(reader.error(DecodeErrorKind::UnexpectedXmlEvent(event))),
+    }
+}
+
+fn read_content(reader: &mut XmlEventReader<impl Read>, tag: &str) -> Result<String, DecodeError> {
+    match reader.expect_next()? {
+        XmlReadEvent::StartElement {
+            name,
+            attributes,
+            namespace,
+        } => {
+            if name.local_name.as_str() != tag {
+                let event = XmlReadEvent::StartElement {
+                    name,
+                    attributes,
+                    namespace,
+                };
+                return Err(reader.error(DecodeErrorKind::UnexpectedXmlEvent(event)));
+            }
+        }
+        event => return Err(reader.error(DecodeErrorKind::UnexpectedXmlEvent(event))),
+    };
+
+    let value = read_content_inner(reader)?;
+
+    reader.expect_end_with_name(tag)?;
+
+    Ok(value)
+}
+
+impl XmlType for Font {
+    const XML_TAG_NAME: &'static str = "Font";
+
+    fn write_xml<W: Write>(&self, writer: &mut XmlEventWriter<W>) -> Result<(), EncodeError> {
+        write_content(writer, &self.family, "Family")?;
+
+        let weight: u16 = self.weight.clone().into();
+        let weight: i32 = weight.into();
```

Comment:
Provided the earlier suggestion for a `to_u16` method is adopted:
```suggestion
        let weight = i32::from(self.weight.to_u16());
```